### ISSUE TITLE
FurTrack.com support

### ISSUE THREAD

**Issue by Eupeptic1** [Simon-Tesla/RaccoonyWebEx#135]
https://FurTrack.com is a site for fursuit photos.  It might be a candidate for Raccoony support.

Someone I follow on another site mentioned that they had uploaded some con photos to FurTrack.  I hadn't heard of it before, so I checked it out.  Basically, it's a site for fursuit photos; they don't accept other types of art.  The submission IDs are up to approximately 92,900, after launching in mid-2019.  Each photo has tags and metadata, and there is a thumbnail and a full-sized photo, similar to other art sites.

It does some single-page-app things; when you are looking at the front page, or at a grid of search results, and you click on one of the images, an overlay pops up with a larger version of the image.  The overlay also has the tags and other metadata, plus `<` and `>` buttons that will take you to the next or previous image in the grid.  The URL does not change when the overlay is up.

It appears to have the idea of a canonical (nikonical?) URL for each image, at `furtrack.com/p/[submission-id]` .  If you visit that URL, you immediately get the overlay with the larger version of the image, tags, metadata, etc, but you *also* get the grid of thumbnails of recent submissions underneath it.

More shortly.

**Comment by Eupeptic1** [comment]
**tl;dr:** I have a plugin that works as well as it can.  FurTrack does some single-page-app things, similar to what Pixiv and FurryNetwork do, that probably require additional support in the rest of Raccoony for the plugin to behave similarly to the plugins for other sites.

I made a first attempt at writing a plugin.  It is on the `experimental-furtrack-support` branch in my repo: https://github.com/Eupeptic1/RaccoonyWebEx/tree/experimental-furtrack-support .

Known issues with this plugin:

1. After the first download, Raccoony is stuck on "Downloaded" (single-page-app issue).

When you're looking at the overlay for a single image, the raccoon head offers the "Download" button, and if you click it, the full-size version of that image, plus the metadata file if requested, gets downloaded.  If you close the overlay and then click on another thumbnail to get its overlay, the raccoon head is still there, but when you mouse over it, the button is stuck on "Downloaded".  You have to reload the page in your browser to get Raccoony to offer the "Download" button again.

This is similar to what happens with the experimental Pixiv support (#26) and the current FurryNetwork support (#117).  Basically, there needs to be some way, either in the main code or the plugins, for Raccoony to detect that things have changed and offer the "Download" button again.

2.  Sometimes Raccoony displays too many buttons (single-page-app issue).

When you're looking at the overlay for a single image, the raccoon-head menu will offer the "Download", "Fullscreen", and "gear" icons (expected), but it will *also* offer the "Open all in tabs" button (unexpected).  The plugin is picking up the grid of other submissions that FurTrack has loaded underneath the overlay.

This may be a feature, not a bug, but it's different than how most of the other plugins work.

3.  Sometimes the plugin gets the wrong direct link for a thumbnail.

The direct link to each submission is not available in the DOM from the thumbnail grid.  I think it's either hiding in a script, or when you click on a thumbnail, FurTrack does a database dip to find out where the full-size image is.

The plugin relies on pulling the submission ID out of the URL for the thumbnail, and converting that to FurTrack's "canonical" URL format.  Unfortunately, sometimes that canonical URL gives you a different picture.

So far, I have only seen this on submission IDs less than 20,000 or so.  I wonder if FurTrack maybe had some deleted images or a database schema change or similar that makes this happen on older submissions.

4. FurTrack offers more metadata than the plugin currently records in the metadata file.

FurTrack pulls information from the EXIF and displays it on the overlay (date photo taken, camera model, shutter speed, F stop, etc).  It also displays a copyright statement, which the photographer/uploader has control over.

The plugin currently doesn't do anything with these "extra" metadata items.  Maybe it would be a good idea to include them somewhere... in the description?  In the tags?  In a new "more data" field?  Dunno.

On the other hand, existing sites like FA and Weasyl display (or used to display?) similar data for photographs, and the plugins for those sites don't currently do anything with that data, so it may be OK just to ignore it.
